engine/collection/synchronization/engine.go

```diff
@@ -372,72 +334,64 @@ CheckLoop:
 		select {
 		case <-e.unit.Quit():
 			break CheckLoop
-		case <-poll.C:
-			err := e.pollHeight()
-			if err != nil {
-				e.log.Error().Err(err).Msg("could not poll heights")
-			}
-
+		case <-pollChan:
+			e.pollHeight()
 		case <-scan.C:
 			final, err := e.state.Final().Head()
 			if err != nil {
-				e.log.Error().Err(err).Msg("could not get final height")
-				continue
+				e.log.Error().Err(err).Msg("could not get last finalized header")
+				return
```

Comment:
should this be fatal if we're going to exit here? Stopping the polling process can cause the node to get permanently out of sync

Reply:
I have replaced it with `continue` but my feeling is that we should crash the node at this point. Not retrieving state seems like a fatal error to me.

Reply:
> my feeling is that we should crash the node at this point

I agree actually, I think it's better to change the log level to `Fatal`